### ISSUE TITLE
Subsequent calls to `recurly.configure()` fail if hosted field(s) didn't fully load

### ISSUE THREAD

**Issue by pjg** [recurly/recurly-js#570]
### Gist of the issue

`recurly.js` sometimes hangs up with the `readyState = 1` and it cannot be easily reset/readded/reconfigured.

### The details

I am able to reproduce this issue in my React app, which allows the user at the checkout page to select different plans. If I click on a different plan, after the hosted CC field is shown as a blank input, but before the CC placeholders show, the parent component re-renders the checkout page, creating a completely new `<div>` placeholder for the Recurly hosted CC fields. However, `recurly.hostedFields.fields[0].target` will still be set and will point to the old `<div>`, before the re-render. This makes subsequent call to `recurly.configure()` fail, even I try to use a different `selector` option for the CC field.

### Debugging details

Here is how the `recurly` object and its children look like when this issue occurs:

![image](https://user-images.githubusercontent.com/10712/69958765-c9a0cd80-1505-11ea-850f-4ed8b49cd50c.png)

![image](https://user-images.githubusercontent.com/10712/69958783-d2919f00-1505-11ea-8495-351fb06ca210.png)

![image](https://user-images.githubusercontent.com/10712/69958824-e2a97e80-1505-11ea-8ba1-37439871e41a.png)

### Workaround attempts

I tried to work around this issue by manually destroying the invalid hosted field and injecting it again, like this:

```javascript
recurly.hostedFields.fields[0].destroy()
recurly.configure({ publicKey: 'key', fields: { card: { selector: '.recurly-cc' } } })
```

Or this:

```javascript
recurly.hostedFields.inject()
```

However, in both approaches I am unable to make this field work again. The CC prompt will not display and the whole CC field will not allow me to type the credit card number.

### Working workaround

```javascript
recurly.readyState = 2
recurly.configure({publicKey: 'key', fields: { card: { selector: '.recurly-cc' } } })
```

This makes the form reset to its initial state and the CC number can be typed again.

### The real fix

It would be really nice if there was an additional check in the `recurly.configure()` call, which would detect this hang-up state and just reset hosted fields.

**Comment by chrissrogers** [comment]
@pjg first, thank you for the excellent detail you've added here. It's made tracing the root causes very straightforward. Yes, I agree with your conclusion to add additional checks for field integrity to `recurly.configure()`. If you have a distilled code sample in mind for reproducing this behavior, I can use that as a litmus for a fix, but as-is, your report will help enormously with the implementation.

Any further detail you'd like to add, do feel free. We'll be looking into this shortly.

**Comment by chrissrogers** [comment]
One thing to note is that the [Elements](https://github.com/recurly/recurly-js/pull/568) feature is designed to work seamlessly with React, and will make integrations like yours much more straightforward. So that is certainly worth tracking once we resolve your use-case here.

Shortly following the release of the Elements feature, we will be releasing a React component as well.

**Comment by pjg** [comment]
I don't have a reproducible testcase, unfortunately. I'm just using a regular React component and I'm triggering a re-rendering of it (link click for the same page/view) _at the moment_ that Recurly starts its initialization process, i.e. when the CC input is blank but before placeholders are being displayed.

**Comment by chrissrogers** [comment]
Hi @pjg -- I wanted to drop a note here about the release of our [react-recurly](https://github.com/recurly/react-recurly) library. It makes use of the elements feature, the spiritual successor to Hosted Fields. We've updated [our documentation](https://developers.recurly.com/reference/recurly-js/) to detail the Elements API, and I think your implementation is a good use case.

Of course, I know that an API upgrade is not always the best option. We're looking into a fix for this issue within Hosted Fields currently, and will provide an update here accordingly.

**Comment by pjg** [comment]
@chrissrogers Thanks! We will probably wait a little bit before switching to Elements until it matures a little bit.

Also thank you for fixing the hosted fields reset :D

**Comment by calebwang** [comment]
We have been experiencing a similar issue and are looking forward to this bugfix. When can we expect version 4.12 to be released?

**Comment by chrissrogers** [comment]
@calebwang hi! We don't have a date specified at this time, but I'll be updating this issue as soon as it's available. Apologies for the longer-than-usual wait!

**Comment by chrissrogers** [comment]
Following up: this fix has been deployed today as part of [`v4.12.0`](https://github.com/recurly/recurly-js/releases/tag/v4.12.0)
